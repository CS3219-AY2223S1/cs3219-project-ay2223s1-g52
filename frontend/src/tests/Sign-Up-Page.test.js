import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import SignUpPage from "views/sign-up-page";
import { Route, BrowserRouter as Router, Routes } from "react-router-dom";

it("renders sign in correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Routes>
          <Route path="/" element={<SignUpPage />} />
        </Routes>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
