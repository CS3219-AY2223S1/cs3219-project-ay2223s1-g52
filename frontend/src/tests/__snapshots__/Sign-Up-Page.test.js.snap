// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sign in correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-1f7x1oc-MuiContainer-root"
>
  <div
    className="MuiBox-root css-yln515"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1tfbpfz-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      variant="outlined"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </button>
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-5el24x-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Create Account
    </h1>
    <form
      className="MuiBox-root css-ti8fc4"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="username"
          id="username-label"
        >
          Username
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="username"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="username"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Username
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Password
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiBox-root css-xdsuyt"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17xdt6p-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled={false}
          id=":r3:"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          />
          Create Account
        </button>
      </div>
    </form>
  </div>
</main>
`;
